pkg/amd/manifest/psp_header.go

```diff
@@ -31,7 +31,46 @@ type PSPHeader struct {
 	Reserved3 [156]byte
 }
 
-// ParsePSPHeader parses
+// Reserved1Offset returns the offset in bytes of field Reserved1
+func (h *PSPHeader) Reserved1Offset() uint64 {
+	return 0
+}
+
+func (h *PSPHeader) Reserved1Length() uint64 {
+	return uint64(binary.Size(h.Reserved1))
+}
+
+// CookieOffset returns the offset in bytes of field Cookie
+func (h *PSPHeader) CookieOffset() uint64 {
+	return h.Reserved1Offset() + h.Reserved1Length()
+}
+
+// CookieLength returns the size in bytes of field Cookie
+func (h *PSPHeader) CookieLength() uint64 {
+	return uint64(binary.Size(h.Cookie))
+}
+
+// Reserved2Offset returns the offset in bytes of field Reserved2
+func (h *PSPHeader) Reserved2Offset() uint64 {
+	return h.CookieOffset() + h.CookieLength()
+}
+
+// Reserved2Length returns the size in bytes of field Reserved2
+func (h *PSPHeader) Reserved2Length() uint64 {
+	return uint64(binary.Size(h.Reserved2))
+}
+
+// VersionOffset returns the offset in bytes of field Version
+func (h *PSPHeader) VersionOffset() uint64 {
+	return h.Reserved2Offset() + h.Reserved2Length()
+}
+
+// VersionLength returns the size in bytes of field Version
+func (h *PSPHeader) VersionLength() uint64 {
+	return uint64(binary.Size(h.Version))
+}
+
+// ParsePSPHeader parses the PSP header that is supposed to be the beginning of each PSP binary
```

Comment:
Somewhere I wrote a tool, which automatically generates that. If relevant I can try to find it.